digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/assetUpdater/LaneUpdater.scala

```diff
@@ -407,12 +409,14 @@ object LaneUpdater {
           GeometryUtils.liesInBetween(replaceInfo.oldToMValue, (originalAdditionalLane.startMeasure, originalAdditionalLane.endMeasure))
       })
       val lanesSplitFromOriginal = replaceInfosAffectingLane.map(replaceInfo => {
-        val newRoadLinkLength = change.newLinks.find(_.linkId == replaceInfo.newLinkId).get.linkLength
+        val newId = if (replaceInfo.newLinkId.nonEmpty) replaceInfo.newLinkId.get else ""
+        val newRoadLinkLength = if (newId.nonEmpty) change.newLinks.find(_.linkId == newId).get.linkLength else 0
```

Comment:
```suggestion
        val newId =  replaceInfo.newLinkId.getOrElse("")
        val links = change.newLinks.find(_.linkId == newId)
        val newRoadLinkLength = if (links.isDefined) links.get.linkLength else 0
```

Seq.Find() on teoreettinen mahdollisuus että linkkiä ei löydy. Olis hyvä tarkistaa että se todellakin löytyi ennen kuin tekee .get.

Reply:
Eikös tuo newId-arvon asetus linkId:ksi tai emptyksi juurikin tarkista linkin olemassaolon ensin? Sitten vielä varmistetaan että pituus haetaan vain if (newId.nonEmpty)

Reply:
joo, toi vähän enemmän tollanen teoreettinen mahdollisuus. ei nyt ehkä tarpeellinen muutos.